### ISSUE TITLE
IllegalArgumentException when inserting to index 0 with non empty DynArray

### ISSUE THREAD

**Issue by Paul-Laxed** [GymnasiumEverstenOldenburg/ADT-Implementierungen#43]
When you try to insert a value to index 0 when the array is not empty the code throws an error

**Comment by GelbEinhalb** [comment]
This happens inside of the `insertAt()` function.

```java
public void insertAt(int index, Type inhalt) {
    if (index == 0 && this.isEmpty()) {
        this.append(inhalt);
        return;
    }
    Item<Type> previous = getItemClass(index - 1);
    Item<Type> newItem = new Item<>(inhalt);

    newItem.setNext(previous.getNext());
    previous.setNext(newItem);

    this.length++;
}
```

we had a similar issue in #14 where @Alexander-Reimer added this piece of code:

```java
if (index == 0 && this.isEmpty()) {
    this.append(inhalt);
    return;
}
```

This solves the case where trying to insert at index 0 when the DynArray is empty would cause an `InvalidArgumentException`.
But not this issue when the DynArray is non-empty.

**Comment by Alexander-Reimer** [comment]
Hopefully closed with #45
